python/pyarrow/__init__.py

```diff
@@ -44,6 +44,25 @@ def parse_version(root):
         __version__ = None
 
 
+try:
+    # We need to do the following to load TensorFlow before
+    # pyarrow.lib. If we don't do this there are symbol clashes
+    # between TensorFlow's use of threading and our global
+    # thread pool, see also
+    # https://issues.apache.org/jira/browse/ARROW-2657 and
+    # https://github.com/apache/arrow/pull/2096.
+    import ctypes
+    import os
+    from sys import platform
+    import site
+    if platform == "linux" or platform == "linux2":
+        SITE_PATH, = site.getsitepackages()
+        ctypes.CDLL(os.path.join(SITE_PATH, "tensorflow",
+                                 "libtensorflow_framework.so"))
+except:
```

Comment:
It might be better to make this less restrictive. You could even get away without a try/except and only attempt to load the shared lib if the TF folder is there. How about putting this in a function in pyarrow.compat?

Reply:
fixed